src/Monolog/Formatter/ElasticsearchLogstashHandler.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+namespace AppBundle\Monitoring\Monolog\Handler;
+
+use Monolog\Formatter\FormatterInterface;
+use Monolog\Formatter\LogstashFormatter;
+use Monolog\Handler\AbstractProcessingHandler;
+use Monolog\Logger;
+use Symfony\Component\HttpClient\HttpClient;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+
+/**
+ * Push logs directly to Elasticsearch and format them according to Logstash specification
+ *
+ * @author Grégoire Pineau <lyrixx@lyrixx.info>
+ */
+class ElasticsearchLogstashHandler extends AbstractProcessingHandler
+{
+    private $endpoint;
+    private $index;
+    private $client;
+    private $responses;
+
+    public function __construct($endpoint = 'http://elasticsearch:9200', $index = 'monolog', HttpClientInterface $client = null, $level = Logger::DEBUG, $bubble = true)
+    {
+        if (!class_exists(HttpClientInterface::class)) {
+            throw new \LogicException(sprintf('%s handler needs symfony/http-client, please run `composer require symfony/http-client`.', __CLASS__));
+        }
+
+        parent::__construct($level, $bubble);
+        $this->endpoint = $endpoint;
+        $this->index = $index;
+        $this->client = $client ?: HttpClient::create();
```

Comment:
If we want to remove the dependency to the component, we can remove this call and make the `$client` argument mandatory. That will only require the http-client-contracts, which is an abstract dependency.

Reply:
we can also keep the argument optional, and throw an exception when the component is missing but not client has been provided, as done in https://github.com/symfony/symfony/blob/4.4/src/Symfony/Component/HttpClient/Psr18Client.php#L65